src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,13 +3,82 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int CAPACITY = 10;
+    private Object[] store;
+    private int position;
+
+    public StorageImpl() {
+        store = new Object[CAPACITY];
+        position = 0;
+    }
 
     @Override
     public void put(K key, V value) {
+        Entry temp;
+        if (key == null) {
+            for (int i = 0; i < position; i++) {
+                temp = (Entry) store[i];
+                if (key == temp.getKey()) {
+                    temp.setValue(value);
+                    store[i] = temp;
+                }
+            }
+            store[position] = new Entry(key, value);
+            position++;
+        } else {
+            for (int i = 0; i < position; i++) {
+                temp = (Entry) store[i];
+                if (key.equals(temp.getKey())) {
+                    temp.setValue(value);
+                    store[i] = temp;
+                }
+            }
+
+            store[position] = new Entry(key, value);
+            position++;
+        }
     }
 
     @Override
     public V get(K key) {
+        Entry temp;
+        if (key == null) {
+            for (int i = 0; i < position; i++) {
+                temp = (Entry) store[i];
+                if (key == temp.getKey()) {
+                    return temp.getValue();
+                }
+            }
+        } else {
+            for (int i = 0; i < position; i++) {
+                temp = (Entry) store[i];
+                if (key == temp.getKey() || key.equals(temp.getKey())) {
+                    return temp.getValue();
+                }
+            }
+        }
         return null;
     }
+
+    public class Entry {
```

Comment:
```suggestion
    private static class Entry<K, V> {
```
aaand now you can create an array I told you about :)

Reply:
Oh! Thanks a lot! Its change everything! :)